src/components/container.jsx

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+var React = require('react');
+var componentMixin = require('./mixins/componentMixin');
+var multiMixin = require('./mixins/multiMixin');
+var FormioComponent = require('../FormioComponent');
+
+module.exports = React.createClass({
+    displayName: 'Container',
+    mixins: [componentMixin, multiMixin],
+    getSingleElement: function(value, index) {
+        return(
+            <div>
+            {this.props.component.components.map(function(component) {
+                    var value = (this.props.data && this.props.data.hasOwnProperty(component.key) ? this.props.data[component.key] : component.defaultValue || '');
+                    var key = (component.key) ? component.key : component.type;
+                    return (
+                        <FormioComponent
+                            {...this.props}
+                            key={key}
+                            name={component.key}
+                            component={component}
+                            value={value}
```

Comment:
Datagrids and containers are two special components that treat their data differently. Because of that we have to treat their onChange event differently.

With most fields they set their value on the root data object like this:
{
  data: {
    firstname: 'Joe'
  }
}

However, datagrids are an array of objects with each component under it being in the array object so you can do this:
{
  data: {
    users: [
      {
        firstname: 'Joe',
        lastname: 'Smith
      },
      {
        firstname: 'Frank',
        lastname: 'Kelly
      }
    ]
  }
}

Containers allow setting the components within an object on that key for example:
{
  data: {
    user: {
      firstname: 'Frank',
      lastname: 'Kelly
    }
  }
}

These can also be nested within each other allowing people to support very complex data structures.

We need to implement something like I did in datagrids only without the array and rows but setting the container to an object that takes the values of each child element and sets its key on the object and passes the container up to the parent onChange event.
https://github.com/formio/react-formio/blob/master/src/components/datagrid.jsx#L72